src/blockchain/staking/history.ts

```diff
@@ -0,0 +1,68 @@
+import { Block, Log } from '@ethersproject/abstract-provider'
+import { BaseProvider } from '@ethersproject/providers'
+import { Abi } from '@xtoken/abis'
+import { formatUnits, Interface } from 'ethers/lib/utils'
+
+import { IStakeHistory } from '../../types/xToken'
+
+import { getXtkStakingContract } from './helper'
+
+const STAKING_HISTORY_START_BLOCK = 12838146
+
+const getHistory = async (
+  provider: BaseProvider,
+  account: string,
+  type: 'stake' | 'unstake',
+  // eslint-disable-next-line functional/no-return-void
+  onError?: (err: Error) => void
+): Promise<readonly IStakeHistory[]> => {
+  const stakingContract = await getXtkStakingContract(provider)
+  const transactionName = type === 'stake' ? 'Stake' : 'UnStake'
+  const label = type === 'stake' ? 'Stake' : 'Unstake'
```

Comment:
Create an utility method called `toTitleCase()` in `src/utils.ts`, you can use `toTitleCase(type)` for `label`
```
export const toTitleCase = (text: string) => {
  return text[0].toUpperCase() + text.slice(1).toLowerCase()
}
```